src/ol/source/ImageArcGISRest.js

```diff
@@ -151,81 +141,24 @@ class ImageArcGISRest extends ImageSource {
     if (this.url_ === undefined) {
       return null;
     }
-
-    resolution = this.findNearestResolution(resolution);
-    pixelRatio = this.hidpi_ ? pixelRatio : 1;
-
-    const image = this.image_;
-    if (
-      image &&
-      this.renderedRevision_ == this.getRevision() &&
-      image.getResolution() == resolution &&
-      image.getPixelRatio() == pixelRatio &&
-      containsExtent(image.getExtent(), extent)
-    ) {
-      return image;
```

Comment:
I wonder if this change will address #14990.

Reply:
I'm sorry, but this change does not address #14990.

Reply:
With the assistance of ahocevar, I have confirmed that the issue #14990 is caused by the limitations of the maxImageHeight and maxImageWidth properties in ArcGIS MapServer. Their default size is set to 4096.